### PR TITLE
coprocessor: support unique index scan

### PR DESCRIPTION
**Pull request by hanfei1991**
Use get interface when scanning unique index.
@AndreMouche PTAL

### CODE REVIEW

File: src/coprocessor/dag/executor/index_scan.rs [hunk 1]
```diff
@@ -289,6 +322,41 @@ mod test {
         assert!(scanner.next().unwrap().is_none());
     }
 
+    #[test]
+    fn test_unique_index_scan() {
+        let mut wrapper = IndexTestWrapper::default();
+
+        let r1 = get_idx_range(TABLE_ID, INDEX_ID, 0, 1); // point get
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/index_scan.rs]: Could you use `prefix_next` to generate point range's `end_key`?

File: src/coprocessor/dag/executor/index_scan.rs [hunk 2]
```diff
@@ -289,6 +322,41 @@ mod test {
...
+            KEY_NUMBER + 2 as i64,
+        ); // point get but miss
+        wrapper.ranges = vec![r1, r2, r3, r4, r5];
+
+        let (snapshot, start_ts) = wrapper.store.get_snapshot();
+        let store = SnapshotStore::new(snapshot, start_ts, IsolationLevel::SI, true);
+        let mut scanner = IndexScanExecutor::new(wrapper.scan, wrapper.ranges, store, true);
+
+        for tid in 0..KEY_NUMBER {
+            let handle = KEY_NUMBER - tid - 1;
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/index_scan.rs]: The result should be order by `handle` ascending  since `desc` in request  is false.

**Comment by AndreMouche** [comment]
@breeswish PTAL

**Comment by AndreMouche** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@AndreMouche Can we merge now?

**Pull request merged**